test/userAgentFilterTests.ts

```diff
@@ -3,21 +3,30 @@
 
 import * as assert from "assert";
 import { WebResource } from "../lib/webResource";
-import { MsRestUserAgentFilter } from "../lib/filters/msRestUserAgentFilter";
+import { MsRestUserAgentPolicy } from "../lib/policies/msRestUserAgentPolicy";
 import { Constants } from "../lib/util/constants";
-import { isNode } from "../lib/msRest";
+import { RequestPolicy, RequestPolicyOptions } from "../lib/policies/requestPolicy";
+import { HttpOperationResponse } from "../lib/httpOperationResponse";
+import { isNode } from "../lib/util/utils";
 
 const should = require("should");
 const userAgentHeader = Constants.HeaderConstants.USER_AGENT;
 
+const emptyRequestPolicy: RequestPolicy = {
+  sendRequest(request: WebResource): Promise<HttpOperationResponse> {
+    assert(request);
+    throw new Error("Not Implemented");
+  }
+};
+
 describe("ms-rest user agent filter (nodejs only)", () => {
   it("should construct user agent header when supplied empty array", function (done) {
     if (!isNode) {
       this.skip();
     }
 
     const userAgentArray: Array<string> = [];
-    const userAgentFilter = new MsRestUserAgentFilter(userAgentArray);
+    const userAgentFilter = new MsRestUserAgentPolicy(emptyRequestPolicy, new RequestPolicyOptions(), userAgentArray);
     const resource = new WebResource();
     resource.headers = {};
     userAgentFilter.before(resource).then((resource) => {
```

Comment:
Does this test still make sense if the RequestPolicy is only supposed to have `.sendRequest()` on it?

Reply:
This test makes sense because it's verifying the behavior of MsRestUserAgentPolicy.before(). I think it makes sense to rename the before() and after() methods that exist on specific RequestPolicies. I can do that now.